### PR TITLE
feat(transactor): Cleanup retry policy and handle error cases

### PR DESCRIPTION
**Pull request by calbera**
## Summary by CodeRabbit

- **New Features**
    - Added health check interval configuration for Ethereum client connections.
    - Introduced a new retry policy for transactions, including exponential backoff strategy for retries.
- **Bug Fixes**
    - Fixed a bug in job manager to handle cases when jitter is nil before calculating task retry sleep duration.
    - Enhanced transaction sending and tracking with improved retry mechanism and error handling.
- **Refactor**
    - Updated health check client creation to support custom health check intervals.
    - Refined transaction retry logic, moving from a default policy to more flexible policy implementations.

### CODE REVIEW

File: core/transactor/sender/retry.go [hunk 1]
```diff
@@ -2,18 +2,57 @@ package sender
...
+func NoRetryPolicy(context.Context, *coretypes.Transaction, error) (bool, time.Duration) {
+	return false, backoffStart
+}
+
+// NewExponentialRetryPolicy returns a RetryPolicy that does an exponential backoff until
+// maxRetries is reached. This does not assume anything about whether the specifc tx should be
+// retried.
+func NewExponentialRetryPolicy() RetryPolicy {
+	backoff := backoffStart
+	retries := make(map[common.Hash]int)
```
Comment (hunter-bera) [inline core/transactor/sender/retry.go:35]: add a mutex on map operations to be thread safe

**Review by hunter-bera — DISMISSED** [review]

**Comment by calbera** [comment]
Why is the `string.Contains()` for error equality checking necessary?
image

**Comment by hunter-bera** [comment]
> Why is the `string.Contains()` for error equality checking necessary? image

Do this 
```go
if myErr, ok := err.(*MyError); ok {
}
```

**Review by itsdevbear — APPROVED** [review]
Overall looks good to me utACK

**Pull request merged**